coriolisclient/tests/cli/test_replica_schedules.py

```diff
@@ -0,0 +1,483 @@
+            mock.sentinel.replica, mock.sentinel.id)
+
+
+class ListReplicaScheduleTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Client List Replica Schedule."""
+
+    def setUp(self):
+        self.mock_app = mock.Mock()
+        super(ListReplicaScheduleTestCase, self).setUp()
+        self.replica = replica_schedules.ListReplicaSchedule(
+            self.mock_app, mock.sentinel.app_args)
+
+    @mock.patch.object(lister.Lister, 'get_parser')
+    def test_get_parser(
+        self,
+        mock_get_parser
+    ):
+        result = self.replica.get_parser(mock.sentinel.prog_name)
+
+        self.assertEqual(
+            mock_get_parser.return_value,
+            result
+        )
+        mock_get_parser.assert_called_once_with(mock.sentinel.prog_name)
+
+    @mock.patch.object(replica_schedules.ReplicaScheduleFormatter,
+                       'list_objects')
+    def test_take_action(
+        self,
+        mock_list_objects
+    ):
+        args = mock.Mock()
+        args.replica = mock.sentinel.replica
+        args.hide_expired = mock.sentinel.hide_expired
+        mock_replica_list = mock.Mock()
+        self.mock_app.client_manager.coriolis.replica_schedules.list = \
+            mock_replica_list
+
+        result = self.replica.take_action(args)
+
+        self.assertEqual(
+            mock_list_objects.return_value,
+            result
+        )
+        mock_replica_list.assert_called_once_with(
+            mock.sentinel.replica, hide_expired=mock.sentinel.hide_expired)
+        mock_list_objects.assert_called_once_with(
+            mock_replica_list.return_value)
+
+
+class ReplicaSchedulesTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Replica Schedules."""
+
+    def test_parse_schedule_group_args(self):
+        class CustomMock(mock.MagicMock):
+            def __getattr__(self, name):
+                return None
+        args = CustomMock()
+        args.minute = mock.sentinel.minute
+        args.hour = mock.sentinel.hour
+        expected_result = {
+            "minute": mock.sentinel.minute,
+            "hour": mock.sentinel.hour
+        }
+
+        result = replica_schedules._parse_schedule_group_args(args)
+
+        self.assertEqual(
+            expected_result,
+            result
+        )
+
+    def test_parse_expiration_date(self):
+        value = None
+
+        result = replica_schedules._parse_expiration_date(value)
+
+        self.assertEqual(
+            None,
+            result
+        )
+
+        value = "2099-12-31"
+
+        result = replica_schedules._parse_expiration_date(value)
+
+        self.assertEqual(
+            datetime.datetime(2099, 12, 31, 0, 0),
+            result
+        )
+
+        value = "2099-31-12"
+
+        self.assertRaises(
+            exceptions.CoriolisException,
+            replica_schedules._parse_expiration_date,
+            value
+        )
```

Comment:
Please add a test case for `_add_schedule_group` where you at least check that arg group 'Schedule` is added to the passed `parser`.